src/Utility/CaseInsensitiveDictionary.php

```diff
@@ -76,7 +76,7 @@ public function offsetGet($offset) {
 	 *
 	 * @throws \WpOrg\Requests\Exception On attempting to use dictionary as list (`invalidset`)
 	 *
-	 * @param string $offset Item name
+	 * @param string|null $offset Item name
```

Comment:
This is incorrect. Offset is not allowed to be `null` and an exception is thrown when it is.

This has to do with the fact that `ArrayAccess` is implemented that this class explicitly does not support adding to this class via `$array[] = ...`.

Reply:
> Offset is not allowed to be `null`

Not **allowed**. You see PHPDoc from the point of view of the user/IDE. I see it as it is written in line number 84.
Please tell me what to do. I'm not able to let go technical facts.

Reply:
All right. I know WordPress users. And it is not typehinted as WP supports PHP 5.
It just makes me stop contributing.

Reply:
If I could add the `string` type declaration, I would, but I can't with the current minimum PHP requirements. And even if the minimum PHP version would be upped, I still couldn't, as it would violate the Liskov principle of contravariance for parameters what with the [`ArrayAccess`](https://www.php.net/manual/en/class.arrayaccess.php) interface declaring the parameter as `mixed`.

So, yes, technically `null` is allowed - or rather `mixed` is allowed -, but that goes against the whole point of the class which in reality only supports integer and string keys (as those are the only array keys supported by PHP itself) and is only _intended_ to be used with string keys.

So I'd rather the docs communicate the type we **want** to receive than to get complaints about the Exception being thrown when `null` would be passed.

Reply:
If I could add the `string` type declaration, I would, but I can't with the current minimum PHP requirements. And even if the minimum PHP version would be upped, I still couldn't, as it would violate the Liskov principle of contravariance for parameters what with the [`ArrayAccess`](https://www.php.net/manual/en/class.arrayaccess.php) interface declaring the parameter as `mixed`.

So, yes, technically `null` is allowed - or rather `mixed` is allowed -, but that goes against the whole point of the class which in reality only supports integer and string keys (as those are the only array keys supported by PHP itself) and is only _intended_ to be used with string keys.

So I'd rather the docs communicate the type we **want** to receive than to get complaints about the Exception being thrown when `null` would be passed.